madmom/features/tempo.py

```diff
@@ -512,6 +726,7 @@ def write_tempo(tempi, filename, mirex=False):
     # format as a numpy array
     out = np.array([t1, t2, strength], ndmin=2)
     # write to output
-    np.savetxt(filename, out, fmt='%.2f\t%.2f\t%.2f')
+    if filename is not None:
+        np.savetxt(filename, out, fmt='%.2f\t%.2f\t%.2f')
```

Comment:
Am I missing something here? How does this change add any meaningful functionality?
Please revert the changes in lines 681-730 or convince me that this is really needed. Line 692 needs to go away nonetheless.